sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicy.java

```diff
@@ -98,6 +101,37 @@ public OpenTelemetryHttpPolicy() {
                 .contextWrite(reactor.util.context.Context.of(REACTOR_PARENT_TRACE_CONTEXT_KEY, startSpan(context)));
     }
 
+    @Override
+    public HttpResponse processSynchronously(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if ((boolean) context.getData(DISABLE_TRACING_KEY).orElse(false)) {
+            return next.processSynchronously();
+        }
+
+        Context telemetryContext = startSpan(context);
+        Object agentContext = OpenTelemetrySpanSuppressionHelper.registerClientSpan(telemetryContext);
+        AutoCloseable closeable = OpenTelemetrySpanSuppressionHelper.makeCurrent(agentContext, telemetryContext);
+        HttpResponse responseToBeRecorded = null;
+        RuntimeException exception = null;
+        try {
+            HttpResponse response = next.processSynchronously();
+            responseToBeRecorded = response;
+            return response;
+        } catch (RuntimeException e) {
+            exception = e;
+            if (e instanceof HttpResponseException) {
+                responseToBeRecorded = ((HttpResponseException) e).getResponse();
+            }
+            throw LOGGER.logExceptionAsError(e); // TODO (kasobol-msft) this probably shouldn't log here.
+        } finally {
+            Span span = Span.fromContext(telemetryContext);
+            spanEnd(span, responseToBeRecorded, exception);
+            try {
+                closeable.close();
+            } catch (Throwable ignored) {
+            }
```

Comment:
Though this will be removed by using try-with-resources, we should never just swallow an exception like this without logging it

Reply:
I took this from here https://github.com/Azure/azure-sdk-for-java/blob/68fa9506ff6981f92e5d1c30981c23ee34c21269/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryHttpPolicy.java#L238-L250 .
I think we can add log in both places and reduce the catch scope to Excpetion (what AutoCloseable mentions).